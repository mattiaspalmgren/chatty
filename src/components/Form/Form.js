import React, { Component } from 'react';
import Button from '../Button/Button.js';
import style from './Form.css';

class Form extends Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(e) {
        const target = e.target;
        const value = target.value;
        const name = target.name;
        this.setState({
            [name]: value
        })
    }

    handleSubmit(e) {
        e.preventDefault();
        const { onSubmit } = this.props;
        onSubmit(this.state);
        e.target.reset();

    }

    render() {
        const { title, fields, status } = this.props;
        const buttonText = status == null ? "Submit" : status;
        return (
            <form className={style.form} onChange={this.handleChange} onSubmit={this.handleSubmit}>
                <h2 className={style.title}>{title}</h2>
                {
                    fields.map(field =>
                        <input key={field} className={style.field} type="text" name={field} placeholder={field}/>
                    )
                }
                <Button text={buttonText}/>
            </form>
        );
    }
}

export default Form;