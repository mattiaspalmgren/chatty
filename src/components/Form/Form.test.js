import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form.js';

describe('Form', () => {
    const FormProps = {
        title: 'Test',
        fields: ['Test1', 'Test1'],
        submitText: 'text',
        onSubmit: jest.fn()
    };
    it('should match its snapshot', () => {
        const component = shallow(<Form {...FormProps}/>);
        expect(component).toMatchSnapshot();
    });
    it('should submit', () => {
        const component = shallow(<Form {...FormProps}/>);
        const preventDefault = jest.fn();
        const reset = jest.fn();
        component.find('form').simulate('submit', { preventDefault, target: { reset: reset } });
        expect(FormProps.onSubmit).toBeCalled();
        expect(preventDefault).toBeCalled();
        expect(reset).toBeCalled();
    });
});
