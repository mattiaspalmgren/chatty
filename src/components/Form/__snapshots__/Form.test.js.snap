// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match its snapshot 1`] = `
<form
  className="form"
  onChange={[Function]}
  onSubmit={[Function]}
>
  <h2
    className="title"
  >
    Test
  </h2>
  <input
    className="field"
    key="Test1"
    name="Test1"
    placeholder="Test1"
    type="text"
  />
  <input
    className="field"
    key="Test1"
    name="Test1"
    placeholder="Test1"
    type="text"
  />
  <Button
    text="text"
  />
</form>
`;
